### ISSUE TITLE
Implement basic regions for single image

### ISSUE THREAD

**Issue by pford** [CARTAvis/carta-backend#80]
- basic ROI for single image including point, rectangle, ellipse, and polygon
- import and export ROI file
- ROI statistics
- ROI spectral profile
- ROI histogram

https://docs.google.com/document/d/1IdEh2x7lrsul4kSJ3lzGCiEUyI9gGTX7yVp8BtCKUyw

**Comment by pford** [comment]
Opened PR 107 to merge regions into master, failed on package upload because no space left on device (prior to running tests).

**Comment by pford** [comment]
This will be ready to test after branch _pam/region_updates_ completes build/test and PR.

**Comment by pford** [comment]
Pull request merged, commit  2d44acc on master building now.  Also fixed rectangle pixels and SET_REGION for cursor region id in this PR.

**Comment by kswang1029** [comment]
A quick test suggests that the major and minor axes of ellipse are swapped. Not sure if it is from  frontend or from backend. Below is the case: the bright pixel is 100 and the backend is all 1.
<img width="856" alt="Screen Shot 2019-03-28 at 11 52 34" src="https://user-images.githubusercontent.com/20819712/55128956-53ff8900-5150-11e9-8d9e-251224cc4973.png">
<img width="855" alt="Screen Shot 2019-03-28 at 11 52 19" src="https://user-images.githubusercontent.com/20819712/55128957-53ff8900-5150-11e9-9025-05d303f809f0.png">

**Comment by veggiesaurus** [comment]
For a rotation of zero, the major axis is north-south, correct? @kswang1029 does this look right?
![image](https://user-images.githubusercontent.com/592504/55139311-1eea3b00-513e-11e9-9e87-e551485d09ab.png)

Can confirm that the backend is receiving and correctly parsing `bmaj=1` and `bmin=0.5` and passing them into `LCEllipsoid` constructor as major and minor axes. I think `LCEllipsoid` assumes a different axis orientation though. The comment above the constructor says:
```
// Construct a two dimensional ellipse with theta being the angle from
    // the x-axis to the major axis of the ellipse in radians.
    LCEllipsoid (
    	const Float xcenter, const Float ycenter,
    	const Float majorAxis, const Float minorAxis,
    	const Float theta, const IPosition& latticeShape
    );
```

**Comment by kswang1029** [comment]
@veggiesaurus yes, major axis should be north-south in astronomy world...

**Comment by pford** [comment]
Ellipse angle PR merged in commit d3f992f, ready to test after builds complete.

**Comment by kswang1029** [comment]
Yes, fixed nicely! Case closed.
